app/controllers/application_controller.rb

```diff
@@ -9,12 +9,21 @@ class ApplicationController < ActionController::API
 
   before_action :authenticate
   before_action :set_app_info_headers
-  skip_before_action :authenticate, only: [:cors_preflight]
+  skip_before_action :authenticate, only: [:cors_preflight, :routing_error]
 
   def cors_preflight
     head(:ok)
   end
 
+  def routing_error
+    raise Common::Exceptions::RoutingError.new(params[:path])
+  end
+
+  def action_missing(m, *args, &block)
```

Comment:
I'm just going to remove that -- I think it's maybe for you call an action from another action, or do a redirect_to or something, not entirely sure tbh. 

It's basically method_missing for controllers -- but since method_missing raises the same dumb error that we can't properly handle in application_controller, I thought I would add it here.